pkg/k8s/network_policy_v1beta_test.go

```diff
@@ -87,12 +89,25 @@ func (s *K8sSuite) TestParseNetworkPolicyDeprecated(c *C) {
 	repo.AddList(rules)
 	c.Assert(repo.CanReachRLocked(&ctx), Equals, api.Allowed)
 
+	matchLabels := make(map[string]string)
+	for _, v := range fromEndpoints {
+		matchLabels[fmt.Sprintf("%s.%s", v.Source, v.Key)] = v.Value
```

Comment:
use the `GetExtendedKeyFrom`: `matchLabels[labels.GetExtendedKeyFrom(k)] = v.Value`